MudFocusTrap: Prevent JSException on Disposed Component

jHabjanMXP opened the pull request:
## Issue
```
[Error] (Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost) Unhandled exception in circuit 'Wvm...'.
Microsoft.JSInterop.JSException: Unable to focus an invalid element.
Error: Unable to focus an invalid element.
    at Object.focus (https://localhost:7039/_framework/blazor.server.js:1:32169)
    at https://localhost:7039/_framework/blazor.server.js:1:2890
    at new Promise (<anonymous>)
    at w.beginInvokeJSFromDotNet (https://localhost:7039/_framework/blazor.server.js:1:2847)
    at Jt._invokeClientMethod (https://localhost:7039/_framework/blazor.server.js:1:63689)
    at Jt._processIncomingData (https://localhost:7039/_framework/blazor.server.js:1:61078)
    at Jt.connection.onreceive (https://localhost:7039/_framework/blazor.server.js:1:54719)
    at i.onmessage (https://localhost:7039/_framework/blazor.server.js:1:78667)
   at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
   at MudBlazor.MudFocusTrap.OnAfterRenderAsync(Boolean firstRender)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)
2025-07-02 18:25:40.025 +02:00 [Warning] (Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer) Unhandled exception rendering component: Unable to focus an invalid element.
```
Received above issue when opening and closing dialog fast multiple times.

## Changes
- using `disposed` check, not to start JsInterop processes in SaveFocusAsync and InitializeFocusAsync AFTER the component has already been disposed of


## Type of Changes
<!-- What type of changes does your code introduce? Put an `x` in only one box that applies best: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)

<!-- If you made any visual changes, provide screenshots of before/after. If it has moving parts, please attach a high quality video. -->

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

jHabjanMXP committed:
MudFocusTrap: Wrap JS calls in try-catch

## Changes
- Catching possible exceptions in MudFocusTrap.OnAfterRenderAsync which are thrown by JSInterop

danielchalmers reviewed (commented):
Is it possible to catch a more specific exception than `Exception`?

versile2 reviewed (commented):
I would prefer to see if (_disposed) return; statements than a bunch of try/catch exceptions. Did you try that?

jHabjanMXP committed:
MudFocusTrap: Using `disposed` check before starting JsInterop tasks

## Changes
- Removing the try-catch from 341d6fa28d13a17cede9d5c69728a9b6b3ca1b27
- using `disposed` check, not to start JsInterop processes in SaveFocusAsync and InitializeFocusAsync AFTER the component has already been disposed of

jHabjanMXP committed:
Merge branch 'dev' into MudFocusTrap-wrapping-jsInterop-in-try-catch

jHabjanMXP commented:
@versile2 I adjusted the solution

versile2 commented:
> @versile2 I adjusted the solution

Did this _isDisposed solve the problem like your other solution?

jHabjanMXP commented:
Yes, it prevents the component from crashing by preventing code being executed after the component is destroyed

danielchalmers commented:
Thank you!
